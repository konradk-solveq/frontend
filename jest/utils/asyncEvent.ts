import {act} from '@testing-library/react-native';

const asyncEvent = async (event: any) => {
    await act(async () => {
        await event;
    });
};

export default asyncEvent;
