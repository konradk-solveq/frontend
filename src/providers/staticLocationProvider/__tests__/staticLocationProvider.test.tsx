import {Text, Pressable} from 'react-native';
import React from 'react';

import renderComponent, {
    staticLocationProviderWrapper,
} from '@jestUtils/render';
import StaticLocationProvider, {
    useLocationProvider,
    LocationDataContext,
} from '@providers/staticLocationProvider/staticLocationProvider';
import asyncEvent from '@jestUtils/asyncEvent';
import {fireEvent} from '@testing-library/react-native';

const customRender = (
    children: React.ReactNode,
    {providerProps, ...renderOptions}: any,
) => {
    return staticLocationProviderWrapper(
        <LocationDataContext.Provider {...providerProps}>
            {children}
        </LocationDataContext.Provider>,
        renderOptions,
    );
};

const locationData = {
    latitude: 53.008773556173104,
    longitude: 20.89136063395526,
};

interface TestComponentIProps {
    started?: boolean;
}
const TestComponent: React.FC<TestComponentIProps> = ({
    started,
}: TestComponentIProps) => {
    const {location, isTrackingActivatedHandler} = useLocationProvider();

    const onPressHandler = () => {
        isTrackingActivatedHandler(!!started);
    };

    return (
        <>
            <Text testID="testComponentText">{JSON.stringify(location)}</Text>
            <Pressable testID="testComponentButton" onPress={onPressHandler} />
        </>
    );
};

describe('<StaticLocationProvider />', () => {
    describe('Rendering', () => {
        it('Should render provider', async () => {
            const provider = await asyncEvent(
                staticLocationProviderWrapper(<StaticLocationProvider />),
            );

            expect(provider).toBeTruthy();
        });
    });

    it('Should render component with provider', async () => {
        const component = await asyncEvent(renderComponent(<TestComponent />));

        expect(component).toBeTruthy();
        expect(component).toMatchSnapshot();
    });

    it("Should render component with provider's values", async () => {
        const onPress = jest.fn().mockImplementationOnce((a: boolean) => {});

        const providerProps = {
            value: {
                location: locationData,
                isTrackingActivatedHandler: onPress,
            },
        };
        const {getByText, getByTestId} = await asyncEvent(
            customRender(<TestComponent />, {providerProps}),
        );

        const textElement = getByText(JSON.stringify(locationData));

        expect(textElement).toBeTruthy();

        const button = getByTestId('testComponentButton');
        fireEvent.press(button);

        expect(onPress).toHaveBeenCalledTimes(1);
    });
});
