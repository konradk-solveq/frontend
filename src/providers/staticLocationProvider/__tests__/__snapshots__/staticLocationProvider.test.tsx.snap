// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaticLocationProvider /> Should render component with provider 1`] = `
Array [
  <Text
    testID="testComponentText"
  />,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="testComponentButton"
  />,
  <View
    style={
      Array [
        Object {
          "position": "absolute",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#313131",
          "bottom": 0,
          "height": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
          "zIndex": 1000,
        }
      }
    >
      <Text
        collapsable={false}
        nativeID="animatedComponent"
        numberOfLines={1}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "DIN2014Narrow-Regular",
            "fontSize": 13,
            "letterSpacing": 0.4,
            "opacity": 0,
            "paddingBottom": 7.444196428571429,
            "paddingHorizontal": 36.231884057971016,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
